pkg/agent/flowexporter/utils.go

```diff
@@ -27,3 +33,64 @@ func NewConnectionKey(conn *Connection) ConnectionKey {
 		strconv.FormatUint(uint64(conn.TupleOrig.Protocol), 10),
 	}
 }
+
+// ParseFlowCollectorAddr parses the flow collector address input for flow exporter and aggregator
+func ParseFlowCollectorAddr(addr string, defaultPort string, defaultProtocol string) (string, string, string, error) {
+	var strSlice []string
+	var host, port, proto string
+	match, err := regexp.MatchString("\\[.*\\]:.*", addr)
+	if err != nil {
+		return host, port, proto, fmt.Errorf("Failed to parse FlowCollectorAddr: %s", addr)
+	}
+	if match {
+		idx := strings.Index(addr, "]")
+		strSlice = append(strSlice, addr[:idx+1])
+		strSlice = append(strSlice, strings.Split(addr[idx+2:], ":")...)
+	} else {
+		strSlice = strings.Split(addr, ":")
+	}
+	if len(strSlice) == 3 {
+		host = strSlice[0]
+		if strSlice[1] == "" {
+			port = defaultPort
+		} else {
+			port = strSlice[1]
+		}
+		if (strSlice[2] != "udp") && (strSlice[2] != "tcp") {
+			return host, port, proto, fmt.Errorf("connection over %s transport proto is not supported", strSlice[2])
+		}
+		proto = strSlice[2]
+	} else if len(strSlice) == 2 {
+		host = strSlice[0]
+		port = strSlice[1]
+		proto = defaultProtocol
+	} else if len(strSlice) == 1 {
+		host = strSlice[0]
+		port = defaultPort
+		proto = defaultProtocol
+	} else {
+		return host, port, proto, fmt.Errorf("flow collector address is given in invalid format")
+	}
+	return host, port, proto, nil
+}
+
+// ParseFlowIntervalString parses the flow poll or export interval input string for flow exporter and aggregator
+func ParseFlowIntervalString(intervalString string) (time.Duration, error) {
+	flowInterval, err := time.ParseDuration(intervalString)
+	if err != nil {
+		return 0, fmt.Errorf("Flow interval string is not provided in right format")
+	}
+	if flowInterval < time.Second {
+		return 0, fmt.Errorf("Flow interval should be greater than or equal to one second")
```

Comment:
we do not typically capitalize error strings

Reply:
Thanks, addressed.